### PR TITLE
Migrated livetvchannels to es6 module

### PR DESCRIPTION
**Pull request by matjaz321**
**Changes**
Migrated `livetvchannels.js` to es6 module
**Issues**
Part of #868

### CODE REVIEW

File: src/controllers/livetv/livetvchannels.js [hunk 1]
```diff
@@ -1,133 +1,134 @@
...
-        }
-
-        function getQuery() {
-            return getPageData().query;
+import cardBuilder from 'cardBuilder';
+import imageLoader from 'imageLoader';
+import libraryBrowser from 'libraryBrowser';
+import loading from 'loading';
+import events from 'events';
+import userSettings from 'userSettings';
```
Comment (Camc314) [inline src/controllers/livetv/livetvchannels.js:6]:
```suggestion
import * as userSettings from 'userSettings';
```

File: src/controllers/livetv/livetvchannels.js [hunk 2]
```diff
@@ -1,133 +1,134 @@
...
+        isLoading = true;
+        const query = getQuery();
+        const apiClient = ApiClient;
+        query.UserId = apiClient.getCurrentUserId();
+        apiClient.getLiveTvChannels(query).then(function (result) {
+            renderChannels(context, result);
+            loading.hide();
+            isLoading = false;
+
+            require(['autoFocuser'], function (autoFocuser) {
```
Comment (Camc314) [inline src/controllers/livetv/livetvchannels.js:118]:
```suggestion
        import('autoFocuser').then(({default: autoFocuser}) => {
```

File: src/controllers/livetv/livetvchannels.js [hunk 3]
```diff
@@ -1,133 +1,134 @@
...
+        isLoading = true;
+        const query = getQuery();
+        const apiClient = ApiClient;
+        query.UserId = apiClient.getCurrentUserId();
+        apiClient.getLiveTvChannels(query).then(function (result) {
+            renderChannels(context, result);
+            loading.hide();
+            isLoading = false;
+
+        import('autoFocuser').then(({default: autoFocuser}) => {
```
Comment (Camc314) [inline src/controllers/livetv/livetvchannels.js:118]:
```suggestion
            import('autoFocuser').then(({default: autoFocuser}) => {
```
Oops, i did the wrong indent

File: src/controllers/livetv/livetvchannels.js [hunk 4]
```diff
@@ -1,133 +1,134 @@
...
-        var pageData;
-        var self = this;
-        var isLoading = false;
-        tabContent.querySelector('.btnFilter').addEventListener('click', function () {
-            showFilterMenu(tabContent);
+    }
+
+    function showFilterMenu(context) {
+        import(['components/filterdialog/filterdialog']).then(({default: filterDialogFactory}) => {
+            const filterDialog = new filterDialogFactory({
```
Comment (MrTimscampi) [inline src/controllers/livetv/livetvchannels.js:95]:
```suggestion
        import(['components/filterdialog/filterdialog']).then(({default: FilterDialog}) => {
            const filterDialog = new FilterDialog({
```

**Pull request merged**